tests/ext/flask_sqlalchemy/test_query.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import absolute_import
+import json
+import jsonpickle
+import pytest
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.context import Context
+from aws_xray_sdk.ext.flask_sqlalchemy.query import XRayFlaskSqlAlchemy
+from flask import Flask
+
+
+app = Flask(__name__)
+app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
+app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///:memory:"
+db = XRayFlaskSqlAlchemy(app)
+
+
+class User(db.Model):
+    __tablename__ = "users"
+
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(255), nullable=False, unique=True)
+    fullname = db.Column(db.String(255), nullable=False)
+    password = db.Column(db.String(255), nullable=False)
+
+
+def _search_entity(entity, name):
+    """Helper function to that recursivly looks at subentities
+    Returns a serialized entity that matches the name given or None"""
+    if 'name' in entity:
+        my_name = entity['name']
+        if my_name == name:
+            return entity
+        else:
+            if "subsegments" in entity:
+                for s in entity['subsegments']:
+                    result = _search_entity(s, name)
+                    if result is not None:
+                        return result
+    return None
+
+
+def find_sub(segment, name):
```

Comment:
I would recommend rename it to find_subsegment. Abbreviate on functions is not a good practice in general if it is not absolutely needed.